OpenKh.Command.IdxImg/Program.cs

```diff
@@ -213,6 +226,49 @@ protected int OnExecute(CommandLineApplication app)
 
                 using var isoStream = File.Open(InputIso, FileMode.Open, FileAccess.ReadWrite);
 
+                // Begin being hacky.
+
+                if (IdxIsoBlock == -1 || ImgIsoBlock == -1)
+                {
+                    byte[] imgNeedle = new byte[] { 0x01, 0x09, 0x4B, 0x48, 0x32, 0x2E, 0x49, 0x4D, 0x47, 0x3B, 0x31 };
+                    byte[] idxNeedle = new byte[] { 0x01, 0x09, 0x4B, 0x48, 0x32, 0x2E, 0x49, 0x44, 0x58, 0x3B, 0x31 };
+
+                    isoStream.Position = (uint)(0x105 * 0x800);
+                    byte[] haystack = isoStream.ReadBytes(0x500);
+
+                    using (BinaryReader reader = new BinaryReader(new MemoryStream(haystack)))
+                    {
+                        for (int i = 0; i < 0x500; i++)
+                        {
+                            reader.BaseStream.Position = i;
+
+                            byte[] hayRead = reader.ReadBytes(0x0B);
+
+                            bool imgCmp = CompareArrays(hayRead, imgNeedle);
+                            bool idxCmp = CompareArrays(hayRead, idxNeedle);
+
+                            if (imgCmp || idxCmp)
+                            {
+                                reader.BaseStream.Position -= 0x24;
+
+                                byte[] blockStack = reader.ReadBytes(4);
+                                byte[] blockCorrect = new byte[] { blockStack[3], blockStack[2], blockStack[1], blockStack[0] };
+
+                                if (idxCmp && IdxIsoBlock == -1)
+                                    IdxIsoBlock = BitConverter.ToInt32(blockCorrect);
```

Comment:
I never considered to use `BitConverter.ToInt32`. Thanks for sharing this tip :)